rhizome/host/lib/vm_setup.rb

```diff
@@ -234,28 +233,75 @@ def routes4(ip4, ip4_local, nics)
     end
   end
 
-  def write_nat4_config(ip4, nics)
+  def write_nftables_conf(ip4, gua, nics)
+    nat4_rules = generate_nat4_rules(ip4, nics.first[1])
+    nic_priv_ip4_filters = generate_ip4_filter_rules(nics)
+    guest_ephemeral = subdivide_network(NetAddr.parse_net(gua)).first
+    nic_public_ip6_filter = generate_ip6_public_filter(nics.first, guest_ephemeral)
+    nic_priv_ip6_filters = generate_ip6_private_filter_rules(nics[1..])
+
+    config = build_nftables_config(nat4_rules, nic_priv_ip4_filters, nic_public_ip6_filter, nic_priv_ip6_filters)
```

Comment:
this is a very long and detailed parameter list to not use keywords, but perhaps it's so tightly coupled it's nabd. It may indicate a separate class.

Reply:
I will refactor this file with structs soon. I will also make this a part of that work.